### ISSUE TITLE
Add alternative screenshots to build process

### ISSUE THREAD

**Issue by mattpen** [phetsims/chipper#825]
In the new sim page design, there is a requirement to use the alternative screenshots that have been added to the sim repos.  Before we can implement this, we'll need to add a processing step to add these images to the build artifacts.  This will require a maintenance release that will affect all sims.

I noted that the naming for the alt screenshots is standardized which should make this relatively straightforward.  However, there is a non-standard number of alternative screenshots (0-3).  Is this intentional or do we need to finish supplying the alternative screenshots before the rest of this work?

**Comment by mattpen** [comment]
Discussed variation in number of images with @amanda-phet - she said this was acceptable and is accounted for in the sim page layout.

There was one image with a typo in the name that I fixed (make-a-ten alt 2).

@ariel-phet said we should try to schedule this before the holidays or 1st week of January due to lower traffic with the school break.

Tagging @jonathanolson so he is aware that I'm going to be working on a batch maintenance release.

**Comment by jonathanolson** [comment]
Should I be involved with the batch process at all, or do you feel comfortable following the process on your own?

**Comment by mattpen** [comment]
@ariel-phet mentioned that since this would affect every sim, but would not actually change the sims themselves, that this would trigger a lot of unnecessary updates in the mobile apps.  We agreed it would be good to find a way to make these available without a production deploy.

@jonathanolson and I discussed the problem over Zoom.

To clarify, the problem is that we would like to have the alternate screenshots (i.e. `/assets/{{sim-name}-alt{{N}}.png`) deployed to the same location and with the same compression/optimization steps as the main/original screenshot.  We would like to accomplish this without forcing every mobile device with our app to download and store every sim, as nothing in the sim html file is changing.

JO proposed that we complete the change to the build process, then complete a batch maintenance release to add this to all existing maintenance branches but discontinue the maintenance release before the production deploy step.  Instead of the production deploy, we would write a one-time script to add the images to the existing production sim directories in place.

One problem with this method is that the existing latest production sims would have artifacts that are not reproducible.  This seems acceptable to me, but I'm not sure what all could be affected by this.

We discussed a couple of other methods to deploy these screenshots. 

  - We could save them in the website repo instead of the sim repo so they could be served directly by tomcat.  
  - Another option would be to write a new deploy process for screenshots that would put them in a different location from the deployed sim.  

Both of these options would decouple the screenshots from the sim deploy process.  This might be beneficial though as we went through this same problem when needing to deploy the twitter card optimizations for the screenshots.

Adding to dev meeting to see if anyone has opinions on Jon's proposal and/or if we should decouple screenshot deployment from sim deployment.

**Comment by pixelzoom** [comment]
12/19/19 dev meeting:

This is HTML5 only.

@mattpen said it would be nice to decouple screenshot deployment from sim deployment.  But it's also nice to have them coupled, so that screenshots go to the right place when deploying a sim. 

Make an independent grunt task to publish screenshots. Include that task as part of the sim-deploy task.  This achieves both goals.

**Comment by mattpen** [comment]
Jimp has stopped working for me in a grunt task, I can't continue on this at the moment: https://github.com/oliver-moran/jimp/issues/852

**Comment by pixelzoom** [comment]
Please ignore the above commits. They should have been tagged for https://github.com/phetsims/sun/issues/575.

**Comment by mattpen** [comment]
Sharp seems worth evaluating:

https://github.com/lovell/sharp
https://medium.com/on-discord/nodejs-image-manipulation-libraries-43a3f955cc67

**Comment by pixelzoom** [comment]
Oh man, I did it again.  I must have this issue stuck on my clipboard.  Please ignore the above commit, it was meant for phetsims/sun#575.

**Comment by mattpen** [comment]
Sharp is working well in local builds but failing on RHEL7.  There is a fix in the pipeline marked as "ready-to-ship": https://github.com/lovell/sharp/issues/2119.

**Comment by mattpen** [comment]
sharp 0.25.2 is now available

**Comment by mattpen** [comment]
The new sharp package had additional images, so I rolled back to using jimp.  It is working nicely, but we need to add a call to the build server to deploy images after a production sim is deployed.

**Comment by mattpen** [comment]
4efc612af24454cd62bd6afeef2e5b4babb438b5 - This is the change to deploy images after a production sim is deployed.
